pkg/collector/corechecks/snmp/checkconfig/config.go

```diff
@@ -143,8 +160,19 @@ func (c *CheckConfig) addUptimeMetric() {
 // warning: changing GetStaticTags logic might lead to different deviceID
 // GetStaticTags does not contain tags from instance[].tags config
 func (c *CheckConfig) GetStaticTags() []string {
-	tags := []string{"snmp_device:" + c.IPAddress}
-	tags = append(tags, c.ExtraTags...)
+	tags := common.CopyStrings(c.ExtraTags)
+	if c.IPAddress != "" {
+		tags = append(tags, "snmp_device:"+c.IPAddress)
+	}
+	return tags
+}
+
+// GetStaticTagsForMetrics return static tags to be used for metrics, service checks, etc
```

Comment:
Why is Network added to static tags only for metrics? Why not for everyone?

Reply:
`GetStaticTagsForMetrics` is used for metrics, service checks, metadata (maybe it shouldn't have "metrics" in the name).

The main reason is that we don't want the network/subnet to be part of the deviceID tags (since we don't want the device ID to change if the subnet change from 10.0.0.0/29 to 10.0.0.0/30 for example).

I refactored the code here 5c8f042a2 to avoid the need for GetStaticTagsForMetrics.

Reply:
Improve the comment here d985dfc85 with:

```
// GetNetworkTags returns network tags
// network tags are not part of the static tags since we don't want the deviceID
// to change if the network/subnet changes e.g. 10.0.0.0/29 to 10.0.0.0/30
```